iOS/ViewControllers/Courses/CourseItemListViewController.swift

```diff
@@ -166,6 +166,45 @@ extension CourseItemListViewController { // TableViewDelegate
         return header
     }
 
+    @available(iOS 13.0, *)
+    override func tableView(_ tableView: UITableView, contextMenuConfigurationForRowAt indexPath: IndexPath, point: CGPoint) -> UIContextMenuConfiguration? {
+        let courseItem = self.dataSource.object(at: indexPath)
+
+        return UIContextMenuConfiguration(identifier: nil, previewProvider: nil) { _ in
+
+            let share = UIAction(title: "Share", image: UIImage(systemName: "square.and.arrow.up")) { _ in
+                courseItem.share(viewController: self)
+            }
+
+            let contextMenu = UIMenu(title: "", children: [share])
+
+            if let video = courseItem.content as? Video {
+
+                if let downloadStream = video.downloadStreamAction, let downloadSlides = video.downloadSlidesAction {
+                    return UIMenu(title: "", children: [share, downloadStream, downloadSlides])
+                }
+
+                if let downloadStream = video.downloadStreamAction {
+                    return UIMenu(title: "", children: [share, downloadStream])
+                }
+
+                if let downloadSlides = video.downloadSlidesAction {
+                    return UIMenu(title: "", children: [share, downloadSlides])
+                }
+
+            }
+
+            return contextMenu
```

Comment:
With the changes of the previous comment, this can be simplified to:
```
let children = [share] + video.actions
return Menu(title:"", children: children)
```

- I think the video actions would be a good fit for a submenu.
- Why is the title blank? When it is not shown, what is the purpose of it? Is it iPad-only?